crypto/rand/rand_unix.c

```diff
@@ -325,6 +329,90 @@ static struct random_device {
 } random_devices[OSSL_NELEM(random_device_paths)];
 static int keep_random_devices_open = 1;
 
+#   if defined(DEVRANDOM_WAIT)
+static void *shm_addr;
+
+#    if !defined(FIPS_MODE)
+static void cleanup_shm(void)
+{
+    shmdt(shm_addr);
+}
+#    endif
+#   endif
+
+/*
+ * Ensure that the system randomness source has been adequately seeded.
+ * This is done by having the first start of libcrypto, read one bytes from
+ * /dev/random.  Subsequent starts of the library and later reseedings do not
+ * need to do this.
+ */
+#   if defined(__linux)
+static int wait_random_seeded(void)
+{
+    static int seeded = 0;
+    static const int kernel_version[] = { DEVRANDOM_SAFE_KERNEL };
+    int kernel[2];
+    int shm_id, fd = -1, r;
+    char c, *p;
+    fd_set fds;
+    struct utsname un;
+
+    if (!seeded && OPENSSL_RAND_SEED_DEVRANDOM_SHM_ID >= 0) {
```

Comment:
hmm.  this is not going to work if OPENSSL_RAND_SEED_DEVRANDOM_SHM_ID < 0
simplest solution would be to initialize
`seeded = OPENSSL_RAND_SEED_DEVRANDOM_SHM_ID < 0`

Reply:
Good point, will fix.

Reply:
I think, it will work.  If the !seeded succeeds, the >= 0 check will fail and it will fall through.  An optimising compiler will detect this and remove earlier check.

Reply:
The initialisation of seeded seems like a better solution and I'll change.

Reply:
Yes, the problem with the original version was that it would return 0 in that case.

Reply:
Good point, lucky I chose you better alternative :)